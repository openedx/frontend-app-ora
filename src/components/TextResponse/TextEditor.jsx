import React from 'react';
import PropTypes from 'prop-types';

import { TextArea } from '@edx/paragon';
import { useIntl } from '@edx/frontend-platform/i18n';
import messages from './messages';

const TextEditor = ({
  // id,
  value,
  disabled,
  optional,
  onChange,
}) => {
  const { formatMessage } = useIntl();

  return (
    <TextArea
      name="text-response"
      className="textarea-response"
      label={`
        ${formatMessage(messages.yourResponse)} (${formatMessage(optional ? messages.optional : messages.required)})
      `}
      value={value}
      onChange={onChange}
      placeholder={formatMessage(messages.textResponsePlaceholder)}
      disabled={disabled}
    />
  );
};

TextEditor.defaultProps = {
  disabled: false,
  value: '',
  optional: false,
  onChange: () => { },
};

TextEditor.propTypes = {
  // id: PropTypes.string.isRequired,
  // input: PropTypes.shape({
  //   value: PropTypes.string,
  //   name: PropTypes.string,
  //   onChange: PropTypes.func.isRequired,
  // }).isRequired,
  // meta: PropTypes.shape({
  //   touched: PropTypes.bool,
  //   submitFailed: PropTypes.bool,
  //   error: PropTypes.string,
  // }).isRequired,
  disabled: PropTypes.bool,
  value: PropTypes.string,
  optional: PropTypes.bool,
  onChange: PropTypes.func,
};

export default TextEditor;
