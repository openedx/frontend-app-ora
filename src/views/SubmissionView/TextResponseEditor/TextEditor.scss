.textarea-response {
  .form-control {
    height: 500px;
  }
}
