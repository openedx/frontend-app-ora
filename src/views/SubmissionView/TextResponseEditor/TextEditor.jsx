import React from 'react';
import PropTypes from 'prop-types';

import { Form } from '@openedx/paragon';
import { useIntl } from '@edx/frontend-platform/i18n';
import messages from './messages';
import './TextEditor.scss';

const TextEditor = ({
  // id,
  value,
  disabled,
  optional,
  onChange,
  isInValid,
}) => {
  const { formatMessage } = useIntl();

  return (
    <Form.Group isInvalid={isInValid}>
      <Form.Control
        as="textarea"
        name="text-response"
        className="textarea-response"
        label={(
          <h3>
            {formatMessage(messages.yourResponse)} ({formatMessage(optional ? messages.optional : messages.required)})
          </h3>
        )}
        value={value}
        onChange={onChange}
        placeholder={formatMessage(messages.textResponsePlaceholder)}
        disabled={disabled}
      />
      { isInValid && <Form.Control.Feedback type="invalid">{formatMessage(messages.requiredField)}</Form.Control.Feedback>}
    </Form.Group>
  );
};

TextEditor.defaultProps = {
  disabled: false,
  value: '',
  optional: false,
  onChange: () => {},
  isInValid: false,
};

TextEditor.propTypes = {
  // id: PropTypes.string.isRequired,
  // input: PropTypes.shape({
  //   value: PropTypes.string,
  //   name: PropTypes.string,
  //   onChange: PropTypes.func.isRequired,
  // }).isRequired,
  // meta: PropTypes.shape({
  //   touched: PropTypes.bool,
  //   submitFailed: PropTypes.bool,
  //   error: PropTypes.string,
  // }).isRequired,
  disabled: PropTypes.bool,
  value: PropTypes.string,
  optional: PropTypes.bool,
  onChange: PropTypes.func,
  isInValid: PropTypes.bool,
};

export default TextEditor;
