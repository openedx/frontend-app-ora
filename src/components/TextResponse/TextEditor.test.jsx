import { shallow } from '@edx/react-unit-test-utils';
import TextEditor from './TextEditor';

describe('<TextEditor />', () => {
  const props = {
    optional: true,
    disabled: false,
    value: 'value',
    onChange: jest.fn().mockName('onChange'),
  };

  it('render optional', () => {
    const wrapper = shallow(<TextEditor {...props} />);
    expect(wrapper.snapshot).toMatchSnapshot();

    expect(wrapper.instance.findByType('TextArea')[0].props.label).toContain('Optional');
  });

  it('render required', () => {
    const wrapper = shallow(<TextEditor {...props} optional={false} />);
    expect(wrapper.snapshot).toMatchSnapshot();

    expect(wrapper.instance.findByType('TextArea')[0].props.label).toContain('Required');
  });
});
