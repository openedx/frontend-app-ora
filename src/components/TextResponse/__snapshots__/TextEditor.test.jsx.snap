// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextEditor /> render optional 1`] = `
<TextArea
  className="textarea-response"
  disabled={false}
  label="
        Your response (Optional)
      "
  name="text-response"
  onChange={[MockFunction onChange]}
  placeholder="Enter your response to the prompt above"
  value="value"
/>
`;

exports[`<TextEditor /> render required 1`] = `
<TextArea
  className="textarea-response"
  disabled={false}
  label="
        Your response (Required)
      "
  name="text-response"
  onChange={[MockFunction onChange]}
  placeholder="Enter your response to the prompt above"
  value="value"
/>
`;
